src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerActionCounts.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Contains the counts of registered actions for an analyzer.
+    /// </summary>
+    public class AnalyzerActionCounts
```

Comment:
@JohnHamby @srivatsn We need to discuss this type at the design meeting. We need to expose something publically so the IDE can log telemetry about number of actions of each kind registered by analyzers.